Shepherd: Unknown event 'judge_retry' not in milestone events

## Problem

During shepherd orchestration, the milestone reporting system fails when trying to report `judge_retry` events:

```
[16:20:42] [ERROR] Unknown event 'judge_retry'
```

## Root Cause

The shepherd CLI attempts to report `judge_retry` milestones at `cli.py:574,652`:

```python
ctx.report_milestone("judge_retry", attempt=judge_retries, ...)
```

But `judge_retry` is not in the `VALID_EVENTS` frozenset in `milestones.py:29-42`. Valid events are:
- started, phase_entered, phase_completed, worktree_created, first_commit, pr_created, heartbeat, completed, blocked, error

## Fix Options

1. **Add to VALID_EVENTS**: Add `"judge_retry"` to the frozenset
2. **Use existing event**: Use `"error"` with `will_retry=True` flag to match existing schema patterns

## Files

- `loom-tools/src/loom_tools/milestones.py:29-42` - VALID_EVENTS definition
- `loom-tools/src/loom_tools/shepherd/cli.py:574,652` - report_milestone calls

## Acceptance Criteria

- [ ] Judge retries are properly reported as milestones
- [ ] No "Unknown event" errors during shepherd orchestration

--- a/loom-tools/src/loom_tools/milestones.py
+++ b/loom-tools/src/loom_tools/milestones.py
@@ -6,7 +6,7 @@ entry point (``main()``) registered as ``loom-milestone``.
 Events
 ------
 started, phase_entered, phase_completed, worktree_created, first_commit,
-pr_created, heartbeat, completed, blocked, error
+pr_created, heartbeat, completed, blocked, error, judge_retry
 """
 
 from __future__ import annotations
@@ -38,6 +38,7 @@ VALID_EVENTS = frozenset(
         "completed",
         "blocked",
         "error",
+        "judge_retry",
     }
 )
 
@@ -53,6 +54,7 @@ _REQUIRED: dict[str, set[str]] = {
     "completed": set(),
     "blocked": {"reason"},
     "error": {"error"},
+    "judge_retry": {"attempt"},
 }
 
 
@@ -105,6 +107,12 @@ def _build_milestone_data(event: str, **kwargs: Any) -> dict[str, Any]:
     elif event == "error":
         data["error"] = kwargs["error"]
         data["will_retry"] = bool(kwargs.get("will_retry", False))
+    elif event == "judge_retry":
+        data["attempt"] = int(kwargs["attempt"])
+        if "max_retries" in kwargs:
+            data["max_retries"] = int(kwargs["max_retries"])
+        if "reason" in kwargs:
+            data["reason"] = kwargs["reason"]
     return data
 
 
@@ -253,6 +261,8 @@ def _log_event(event: str, data: dict[str, Any]) -> None:
         log_warning(f"Blocked: {data['reason']}")
     elif event == "error":
         log_error(f"Error: {data['error']}")
+    elif event == "judge_retry":
+        log_warning(f"Judge retry: attempt {data['attempt']}")
 
 
 # ── CLI ─────────────────────────────────────────────────────────
@@ -275,6 +285,7 @@ Events:
   completed           --task-id ID [--pr-merged]
   blocked             --task-id ID --reason "reason" [--details "details"]
   error               --task-id ID --error "message" [--will-retry]
+  judge_retry         --task-id ID --attempt NUM [--max-retries NUM] [--reason "reason"]
 
 Examples:
   loom-milestone started --task-id abc1234 --issue 42 --mode force-pr
@@ -283,6 +294,7 @@ Examples:
   loom-milestone heartbeat --task-id abc1234 --action "running tests"
   loom-milestone completed --task-id abc1234 --pr-merged
   loom-milestone error --task-id abc1234 --error "build failed" --will-retry
+  loom-milestone judge_retry --task-id abc1234 --attempt 1 --max-retries 3 --reason "no review submitted"
 """,
     )
 
@@ -320,6 +332,12 @@ Examples:
     parser.add_argument("--error", dest="error_msg", help="Error message (for 'error')")
     parser.add_argument(
         "--will-retry", action="store_true", help="Error is recoverable (for 'error')"
+    )
+    parser.add_argument(
+        "--attempt", type=int, help="Retry attempt number (for 'judge_retry')"
+    )
+    parser.add_argument(
+        "--max-retries", type=int, help="Maximum retries allowed (for 'judge_retry')"
     )
     parser.add_argument(
         "--quiet", "-q", action="store_true", help="Suppress output on success"
@@ -371,6 +389,10 @@ Examples:
         kwargs["error"] = args.error_msg
     if args.will_retry:
         kwargs["will_retry"] = True
+    if args.attempt is not None:
+        kwargs["attempt"] = args.attempt
+    if args.max_retries is not None:
+        kwargs["max_retries"] = args.max_retries
 
     ok = report_milestone(
         repo_root, args.task_id, args.event, quiet=args.quiet, **kwargs

--- a/loom-tools/tests/test_milestones.py
+++ b/loom-tools/tests/test_milestones.py
@@ -447,3 +447,84 @@ class TestOutputFormat:
         report_milestone(repo, "abc1234", "error", error="fail", will_retry=True)
         data = _read_progress(repo, "abc1234")
         assert data["milestones"][-1]["data"]["will_retry"] is True
+
+
+# ── judge_retry ─────────────────────────────────────────────────
+
+
+class TestJudgeRetryEvent:
+    def test_records_judge_retry(self, repo: pathlib.Path) -> None:
+        """judge_retry event should record attempt and optional fields."""
+        report_milestone(repo, "abc1234", "started", issue=42)
+        ok = report_milestone(
+            repo,
+            "abc1234",
+            "judge_retry",
+            attempt=1,
+            max_retries=3,
+            reason="no review submitted",
+        )
+        assert ok
+
+        data = _read_progress(repo, "abc1234")
+        milestone = data["milestones"][-1]
+        assert milestone["event"] == "judge_retry"
+        assert milestone["data"]["attempt"] == 1
+        assert milestone["data"]["max_retries"] == 3
+        assert milestone["data"]["reason"] == "no review submitted"
+
+    def test_attempt_only(self, repo: pathlib.Path) -> None:
+        """judge_retry with only required attempt field."""
+        report_milestone(repo, "abc1234", "started", issue=42)
+        ok = report_milestone(repo, "abc1234", "judge_retry", attempt=2)
+        assert ok
+
+        data = _read_progress(repo, "abc1234")
+        milestone = data["milestones"][-1]
+        assert milestone["event"] == "judge_retry"
+        assert milestone["data"]["attempt"] == 2
+        assert "max_retries" not in milestone["data"]
+        assert "reason" not in milestone["data"]
+
+    def test_missing_attempt_returns_false(self, repo: pathlib.Path) -> None:
+        """judge_retry without attempt should fail."""
+        report_milestone(repo, "abc1234", "started", issue=42)
+        ok = report_milestone(repo, "abc1234", "judge_retry")
+        assert not ok
+
+    def test_attempt_is_integer(self, repo: pathlib.Path) -> None:
+        """attempt should be an integer."""
+        report_milestone(repo, "abc1234", "started", issue=42)
+        report_milestone(repo, "abc1234", "judge_retry", attempt=1, max_retries=3)
+        data = _read_progress(repo, "abc1234")
+        assert isinstance(data["milestones"][-1]["data"]["attempt"], int)
+        assert isinstance(data["milestones"][-1]["data"]["max_retries"], int)
+
+    def test_judge_retry_via_cli(
+        self, repo: pathlib.Path, monkeypatch: pytest.MonkeyPatch
+    ) -> None:
+        """judge_retry should work via CLI."""
+        monkeypatch.chdir(repo)
+        clear_repo_cache()
+        main(["started", "--task-id", "abc1234", "--issue", "42", "--quiet"])
+        rc = main(
+            [
+                "judge_retry",
+                "--task-id",
+                "abc1234",
+                "--attempt",
+                "1",
+                "--max-retries",
+                "3",
+                "--reason",
+                "no review submitted",
+                "--quiet",
+            ]
+        )
+        assert rc == 0
+        data = _read_progress(repo, "abc1234")
+        milestone = data["milestones"][-1]
+        assert milestone["event"] == "judge_retry"
+        assert milestone["data"]["attempt"] == 1
+        assert milestone["data"]["max_retries"] == 3
+        assert milestone["data"]["reason"] == "no review submitted"

